app/assets/javascripts/oxalis/model/reducers/skeletontracing_reducer.js

```diff
@@ -543,11 +565,16 @@ function SkeletonTracingReducer(state: OxalisState, action: ActionType): OxalisS
         }
 
         case "SET_TREE_GROUP": {
-          return update(state, {
-            tracing: {
-              skeleton: { trees: { [action.treeId]: { groupId: { $set: action.groupId } } } },
-            },
-          });
+          const treeGroupMap = getTreeGroupsMap(skeletonTracing);
+          if (action.groupId == null || treeGroupMap[action.groupId]) {
```

Comment:
@daniel-wer What do you think about the validation here? I think, it would be good if we ensure that the store does not hold invalid values. However, maybe it's enough if we do this validation in the API code? So, our policy would be "external code is always validated. internal code needs to do its own validation"? That would mean, that I move this validation check into the API..

Reply:
Yes I think that's reasonable. For our internal code, the flow typings somewhat fulfill the validation, I'd say, meaning if our flow types are correct, there shouldn't be any surprises. However, for the api functions, we have no control over what the users supply as parameters, so validation sounds like a good idea there :)

Reply:
Well, flow does only check whether the groupId is a number, but does not check whether the index really exists. But I agree that we should assume that **our** code works™.